src/index.ts

```diff
@@ -1,102 +1,224 @@
-    let pushMsg = `Pushing ${imageToPush} to ${registry}`;
-    if (username) {
-        pushMsg += ` as ${username}`;
+async function getPodmanPath(): Promise<string> {
+    if (podmanPath == null) {
+        podmanPath = await io.which("podman", true);
     }
-    core.info(pushMsg);
 
-    //check if images exist in podman's local storage
-    const checkImages = await execute(podman, ['images', '--format', 'json']);
+    return podmanPath;
+}
 
-    const parsedCheckImages = JSON.parse(checkImages.stdout);
+// base URL that gets appended if image is pulled from the Docker imaege storage
+const dockerBaseUrl = "docker.io/library";
 
-    // this is to temporarily solve an issue with the case-sensitive of the property field name. i.e it is Names or names??
-    const nameKeyMixedCase = parsedCheckImages[0] && Object.keys(parsedCheckImages[0]).find(key => 'names' === key.toLowerCase());
-    const imagesFound = parsedCheckImages.
-                            filter((image: string) => image[nameKeyMixedCase] && image[nameKeyMixedCase].find((name: string) => name.includes(`${imageToPush}`))).
-                            map((image: string ) => image[nameKeyMixedCase]);
+async function run(): Promise<void> {
+    const imageInput = core.getInput("image", { required: true });
+    const tag = core.getInput("tag") || "latest";
+    const registry = core.getInput("registry", { required: true });
+    const username = core.getInput("username", { required: true });
+    const password = core.getInput("password", { required: true });
+    const tlsVerify = core.getInput("tls-verify");
+    const digestFileInput = core.getInput("digestfile");
+
+    let imageToPush = `${imageInput}:${tag}`;
+
+    // check if image exist in Podman image storage
+    const isPresentInPodman: boolean = await checkImageInPodman(
+        imageToPush,
+    );
+
+    // check if image exist in Docker image storage and if exist pull the image to Podman
+    const isPresentInDocker: boolean = await pullImageFromDocker(
+        imageToPush,
+    );
 
-    if (imagesFound.length === 0) {
-        //check inside the docker daemon local storage
-        await execute(podman, [ 'pull', `docker-daemon:${imageToPush}` ]);
+    // failing if image is not found in Docker as well as Podman
+    if (!isPresentInDocker && !isPresentInPodman) {
+        throw new Error(`Image ${imageToPush} not found in Podman local storage, or Docker local storage.`);
+    }
+
+    // boolean value to check if pushed image is from Docker image storage
+    let isPushingDockerImage = false;
+
+    if (isPresentInPodman && isPresentInDocker) {
+        const isPodmanImageLatest = await isPodmanLocalImageLatest(
+            imageToPush,
+        );
+
+        if (!isPodmanImageLatest) {
+            core.warning(`The version of ${imageToPush} in the Docker image storage is more recent than the version in the Podman image storage. The image from the Docker image storage will be pushed.`);
```

Comment:
Can you split these up over multiple lines for developer readability?

The user should still see it as one line.